proposals/build-provenance-and-code-signing-for-homebrew.md

```diff
@@ -0,0 +1,174 @@
+
+The proposal is broken up into individual technical items, which are meant to be semi-independent units of work. The units of work are ordered by dependency relationship; i.e. signature generation must come before any signatures can be verified on the client side.
+
+## Technical Background
+
+[Homebrew](https://brew.sh/) is the predominant package manager for macOS, with millions of daily users and [hundreds of active contributors](https://github.com/Homebrew/brew/graphs/contributors). Homebrew is written in Ruby, and uses GitHub Actions to create binary packages (called "[bottles](https://docs.brew.sh/Bottles)") of popular open source projects, which clients then install on their machines via the `brew` CLI. Homebrew successfully delivers [over 400 million](https://formulae.brew.sh/analytics/install/365d/) binary builds of open source packages to macOS users each year.
+
+[Sigstore](https://www.sigstore.dev/) is a Linux Foundation project aimed at making codesigning available to everybody. It uses a combination of open standards (OpenID Connect, WebPKI, Certificate Transparency) to bind non-cryptographic identities (such as emails or GitHub Actions workflow identities) to short-lived signing keys, which can then be used to sign for artifacts much like `gpg` or any other signing tool.
+
+[SLSA](https://slsa.dev) is a set of incrementally adoptable guidelines for supply chain security, established by industry consensus, as well as a specification for attesting to the usage of these best practices in a verifiable way. The specification set by SLSA is useful for both software producers and consumers: producers can follow SLSA's guidelines to make their software supply chain more secure, and consumers can use SLSA to make decisions about whether to trust a software package.
+
+## Technical Items
+
+### Provenance attestations and signatures for official Homebrew taps
+
+Homebrew's packaging processes are built around GitHub Actions. In particular, Homebrew uses GitHub Actions to perform builds of all core formulae (package specifications, like LLVM, `bash`, or `git`) into bottles (binary distributions that can be installed on client machines, similar to an `.app` or `.pkg`). Bottles are then hosted on GitHub Packages (or other hosts, for third-party taps) and are installed by end-user clients e.g. during `brew install` invocations.
+
+Because Homebrew uses GitHub Actions for all `homebrew-core` builds, the Homebrew CI has trivial access to GitHub Actions' [ambient OpenID Connect identities](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect). These identities can in turn be used to generate build provenance and sign with Sigstore, meaning that Homebrew's existing CI workflows would require relatively little additional configuration (and zero external key management) to produce valid SLSA attestations and Sigstore signatures for each bottle build.
```

Comment:
```suggestion
Because Homebrew uses GitHub Actions for all `homebrew-core` builds, the Homebrew CI has trivial access to GitHub Actions' [ambient OpenID Connect identities](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect). These identities can in turn be used to generate build provenance and sign with Sigstore, meaning that Homebrew's existing CI workflows would require relatively little additional configuration (and zero external key management) to produce valid in-toto attestations and Sigstore signatures for each bottle build.
```